### PR TITLE
[SDPA] Update to sdpa benchmark to take into account pt2.0 stack

### PR DESCRIPTION
**Pull request by drisspg**
Updates to sdp benchmark to fix failures due to sdp being included into nn.f.mha. As well compare against compiled version.

### CODE REVIEW

File: benchmarks/transformer/sdp.py [hunk 1]
```diff
@@ -65,93 +145,203 @@ def build_composite_mha_from_nn_mha(pt):
...
+        ),
+        seq_len_list,
+    )
+
+
+def benchmark_torch_function_in_microseconds(f, *args, **kwargs):
+    t0 = benchmark.Timer(
+        stmt="f(*args, **kwargs)", globals={"args": args, "kwargs": kwargs, "f": f}
+    )
+    return t0.blocked_autorange().mean * 1e6
```
Comment (drisspg) [inline benchmarks/transformer/sdp.py:190]: mean or median give similar results, since timer excludes adding stats during warmup I thought mean was a good choice

**Comment by drisspg** [comment]
These comparisons get trickier and tricker to make with every PR I land lol. With large embeddings sizes where the matmuls dominate the benefits of using the nested tensor are apparent. Performance independent of dtype scales linearly with amount of padding. 

When smaller transfomers are used the dense path for the same parameters can be faster than the nested case. BUTTTT the dense case is not masking out elements. This is because if you call nn.mha with a mask it gets converted under the hood to a nested tensor.  It is not as easy to get "vanilla" pytorch results anymore where you do masked softmax. 

IF you manually turn off the fast path for nn.mha in actications.py the results can compared against.

An example output of this script:
Screen Shot 2022-12-02 at 7 17 27 PM


From what I have seen running torch.compile() has no effect on performance.

**Review by cpuhrsch — APPROVED** [review]
